### PR TITLE
Refactor graph

### PR DESCRIPTION
**Pull request by martinloretzzz**
Originally wanted to include the PRUNEGRAPH into the creation step of the graph, but couldn't get that working(Needed something like the AST of a DeviceBuffer).

Removes a few lines, has a few readability improvements and adds the GRAPHPATH option to set the output location and name of the graph (I like to use ./net)

Produces the same graph for the test_efficientnet.py(except adding a prunable key to the nodes).

### CODE REVIEW

**Comment by geohot** [comment]
Can you add tests for the graph to know it's good to merge? :)

**Comment by martinloretzzz** [comment]
I will add some.

**Comment by martinloretzzz** [comment]
Added the tests and made some changes to make it testable.

File: tinygrad/graph.py [hunk 1]
```diff
@@ -1,76 +1,72 @@
...
   atexit.register(save_graph_exit)
 
-global_num_max = 0
 def nm(x):
-  global global_num_max
-  if not hasattr(x, 'global_num'):
-    setattr(x, 'global_num', global_num_max)
-    global_num_max += 1
-  return f"<{x.global_num}>"
+  if not hasattr(x, 'node_id'):
```
Comment (geohot) [inline tinygrad/graph.py:29]: Err, I wouldn't put this in GlobalCounters. There's no use for this outside the graph, and I don't think reset does something anyone wants. Rest looks good, revert that change and I'll merge

Comment (martinloretzzz) [reply tinygrad/graph.py:31]: Done

**Pull request merged**